managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1294,9 +1293,24 @@ public synchronized void updateLedgersIdsComplete(Stat stat) {
             log.debug("[{}] Resending {} pending messages", name, pendingAddEntries.size());
         }
 
+        // Avoid use same OpAddEntry between different ledger handle
+        int pendingSize = pendingAddEntries.size();
+        OpAddEntry existsOp;
+        do {
+            existsOp = pendingAddEntries.poll();
+            if (existsOp != null) {
+                // If op is used by another ledger handle, we need to close it and create a new one
+                if (existsOp.ledger != null) {
+                    existsOp.close();
```

Comment:
not sure we need to close here. I think once we duplicate the operation, we can just let the original callback close the old op. so it seems to me that we don't need introducing another `state` field here.

Reply:
We need to close the original op, otherwise when the old op callback, will poll the first op in the pendingAddEntries  https://github.com/apache/pulsar/blob/ef23a4b0ef82db67cd6610d6fe82dc7aae1d3414/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java#L165

But, the first op is the new op we replaced.